src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1431,16 +1429,32 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                         foreach (var (docId, cv, modifiedTicks) in pendingAttachmentsTombstoneUpdates)
                         {
                             var doc = context.DocumentDatabase.DocumentsStorage.Get(context, docId, DocumentFields.ChangeVector, throwOnConflict: false);
-                            
-                            // RavenDB-19421: if the document doesn't exist, the tombstone doesn't matter
+
+                            // RavenDB-19421: if the document doesn't exist and a conflict for the document doesn't exist, the tombstone doesn't matter
                             // and if the change vector is already merged, we should also check if we had a previous conflict on the existing document
                             // if not, then it is already taken into consideration
                             // we need to force an update when this is _not_ the case, because this replication batch gave us the tombstone only, without
                             // the related document update, so we need to simulate that locally
-                            if (doc != null &&
-                                (ChangeVectorUtils.GetConflictStatus(cv, doc.ChangeVector) != ConflictStatus.AlreadyMerged 
-                                 || doc.Flags.Contain(DocumentFlags.HasAttachments | DocumentFlags.Resolved)))
-                            {   
+
+                            if (doc == null)
+                            {
+                                var conflicts = database.DocumentsStorage.ConflictsStorage.GetConflictsFor(context, docId);
+                                foreach (var documentConflict in conflicts)
+                                {
+                                    if (documentConflict.Flags.Contain(DocumentFlags.HasAttachments) == false ||
+                                        ChangeVectorUtils.GetConflictStatus(cv, documentConflict.ChangeVector) == ConflictStatus.AlreadyMerged)
+                                        continue;
+
+                                    // recreate attachments reference
+                                    database.DocumentsStorage.AttachmentsStorage.PutAttachmentRevert(context, documentConflict.Id, documentConflict.Doc, out _);
```

Comment:
can it be that you create here more reference than required?

consider you have a conflict that wasn't resolved, so you get here..
then you get another doc which is also in conflict with the others, so you will create the reference twice for the previous conflicts, no?

Reply:
In the worst case, it will overwrite the existing reference because it has the same key

Reply:
In the worst case, we'll overwrite an existing attachment reference but won't get two references of the same attachment. In your scenario, we end up with three document conflicts and two attachments